contrib_bots/lib/commute_bot.py

```diff
@@ -0,0 +1,228 @@
+            ))
+
+    # parameters are for the HTTP request
+    params = {}
+
+    # adds API Authentication Key to url request
+    def get_api_key(self):
+        # path for settings file
+        home = expanduser('~')
+        settings = open(home + '/settings.ini')
+        config = cp.ConfigParser()
+        config.readfp(self.settings)
+        api_key = config.get('Google.com', 'api_key')
+        self.params.update({'key': api_key})
+        self.settings.close()
+
+    def validate_requests(self, request, message, client):
+        if request.status_code == 200:
+            return request.json()
+        else:
+            self.send_info(message,
+                           "Something went wrong. Please try again."
+                           + " Error: {error_num}.\n{error_text}"
+                           .format(error_num=request.status_code,
+                                   error_text=request.text), client)
+            return
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        content_list = original_content.split()
+        params = self.params
+
+        if "help" in content_list:
+            self.send_info(message, self.help_info, client)
+            return
+
+        # converts input parameters for api into a dictionary
+        # structure for more aethestic coding
+        for item in content_list:
+            # enables key value pair
+            org = item.split('=')
+            # ensures that invalid inputs are not entered into url request
+            if len(org) != 2:
+                continue
+            key, value = org
+            params.update({key: value})
+
+        # limited to UTC timezone because of difficulty with user inputting
+        # correct string for input
+        if 'departure_time' in params:
+            if params['departure_time'] != 'now':
+                departure_time = params['departure_time']
+                times = departure_time.split(',')
+                times = [int(x) for x in times]
+                # UNIX time from January 1, 1970 00:00:00
+                unix_start_date = dt.datetime(1970, 1, 1, 0, 0, 0)
+                requested_time = dt.datetime(*times)
+                total_seconds = str(int((requested_time-unix_start_date)
+                                        .total_seconds()))
+                params.update({'departure_time': total_seconds})
+
+        elif 'arrival_time' in params:
+            arrival = params['arrival_time']
+            times = arrival.split(',')
+            times = [int(x) for x in times]
+            # UNIX time from January 1, 1970 00:00:00
+            unix_start_date = dt.datetime(1970, 1, 1, 0, 0, 0)
+            requested_time = dt.datetime(*times)
+            total_seconds = str(int((requested_time-unix_start_date)
+                                    .total_seconds()))
+            params.update({'arrival_time': total_seconds})
+
+        self.get_api_key()
```

Comment:
As per our chat on Zulip - I think this is where the bot crashes on a second request since you're trying to retrieve the API key from a closed file. You only need to get the API key once, could you add some logic here to handle the issue?